From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 9 Jun 2019 13:10:51 +0200
Subject: cpu_map: Define md-clear CPUID bit

CVE-2018-12126, CVE-2018-12127, CVE-2018-12130, CVE-2019-11091

The bit is set when microcode provides the mechanism to invoke a flush
of various exploitable CPU buffers by invoking the VERW instruction.

This is a backport of upstream commit 538d873571d7a682852dc1d70e5f4478f4d64e85
---
 src/cpu/cpu_map.xml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/cpu/cpu_map.xml b/src/cpu/cpu_map.xml
index 7d5540a..1269eb0 100644
--- a/src/cpu/cpu_map.xml
+++ b/src/cpu/cpu_map.xml
@@ -283,6 +283,9 @@
     <feature name='avx512-4fmaps'>
       <cpuid eax_in='0x07' edx='0x00000008'/>
     </feature>
+    <feature name='md-clear'> <!-- md_clear -->
+      <cpuid eax_in='0x07' edx='0x00000400'/>
+    </feature>
 
     <!-- Processor Extended State Enumeration sub leaf 1 -->
     <feature name='xsaveopt'>
